lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Addresses.hs

```diff
@@ -253,6 +255,59 @@ spec = describe "SHELLEY_ADDRESSES" $ do
                 (`shouldNotSatisfy` T.isPrefixOf "addr_test")
             ]
 
+    it "ADDRESS_LIST_06 - Change addresses are listed even after transaction is no longer pending" $ \ctx -> runResourceT $ do
+        let initPoolGap = 10
+
+        -- 1. create shelley wallet
+        wSrc <- fixtureWallet ctx
+        wDest <- emptyWalletWith ctx ("Wallet", "cardano-wallet", initPoolGap)
+
+        -- 2. list addresses of that wallet (for later comparison)
+        addrs <- listAddresses @n ctx wDest
+        liftIO (length addrs `shouldBe` 10)
+
+        -- 3. send a transaction from that wallet
+        let destination = (addrs !! 0) ^. #id
+        let payload = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{minUTxOValue},
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+
+        _rTrans <- request @(ApiTransaction n) ctx
+            (Link.createTransaction @'Shelley wSrc) Default payload
+
+        -- 4. quickly, list addresses before the transaction is pending.
+        -- See that a new change address is listed there
+        addrsChange <- request @[ApiAddress n] ctx
+            (Link.listAddresses @'Shelley wSrc) Default Empty
+        verify addrsChange
+            [ expectResponseCode HTTP.status200
+            , expectListSize 31
```

Comment:
I'm guessing that 31 is because `fixtureWallet` has address pool gap = 20 and there are funds on first 10 addresses, so initially there was 30 addresses on the wallet, so the change address is generated as next one 31st. Perhaps a comment would be handy explaining this.